### PR TITLE
OJ-27554 repo counting

### PR DESCRIPTION
**Pull request by thingscouldbeworse**

### CODE REVIEW

File: jf_agent/validation.py [hunk 1]
```diff
@@ -93,6 +99,29 @@ def validate_jira(config, creds):
...
+            for project in git_config.git_include_projects:
+                client = None
+                token = creds.git_instance_to_creds.get(cred_slug).get('github_token')
+                try:
+                    client = GithubGqlClient(base_url=git_config.git_url, token=token)
+                    repo_count = client.get_repos_count(login=project)
+                    client.session.close()
+                    msg = f"credentials preface: {cred_slug} " \
+                          f"identified {repo_count} repos in instance {git_config.git_instance_slug}"
+                    agent_logging.log_and_print(logger, logging.WARNING, msg=msg)
```
Comment (gavinpitt-jf) [inline jf_agent/validation.py:114]: Should this be a `logging.INFO` level?

File: jf_agent/main.py [hunk 2]
```diff
@@ -132,6 +132,20 @@ def main():
 
     else:
         jellyfish_endpoint_info = obtain_jellyfish_endpoint_info(config, creds)
+        if jellyfish_endpoint_info.jf_options:
```
Comment (gavinpitt-jf) [inline jf_agent/main.py:135]: NIT:

If you default `jf_options` to an empty dict in `obtain_jellyfish_endpoint_info` you might be able to simplify this block and not do as much null checking. This also might be a good fail safe incase something upstream changes tho

**Review by gavinpitt-jf — APPROVED** [review]
Approve w/ NITs

File: jf_agent/main.py (hunk 2 above)
Comment (thingscouldbeworse) [reply jf_agent/main.py]: swapped here: https://github.com/Jellyfish-AI/jf_agent/pull/248/commits/2bfdbd50c418a9336b588881aebbe9e79c13ce57#diff-b9f32c609e4bd39398d8a39e8f65ae5978f4b9db7a7c023eaa470ee438043373R136

File: jf_agent/validation.py (hunk 1 above)
Comment (thingscouldbeworse) [reply jf_agent/validation.py]: yep, swapped here: https://github.com/Jellyfish-AI/jf_agent/pull/248/commits/2bfdbd50c418a9336b588881aebbe9e79c13ce57#diff-53a011a4f83bc5fb36e2b2215e792e2c0527cda7df39d1cf09c137eb568a7cb0R114

File: jf_agent/main.py [hunk 3]
```diff
@@ -312,9 +326,11 @@ def obtain_jellyfish_endpoint_info(config, creds):
         )
         raise BadConfigException()
 
-    agent_config = resp.json()
-    jira_info = agent_config.get('jira_info')
-    git_instance_info = agent_config.get('git_instance_info')
+    agent_config_from_api = resp.json()
+    jira_info = agent_config_from_api.get('jira_info')
+    git_instance_info = agent_config_from_api.get('git_instance_info')
+    jf_options = agent_config_from_api.get("jf_options")
```
[the remark this answers is not in the document]
Comment (thingscouldbeworse) [reply jf_agent/main.py]: https://github.com/Jellyfish-AI/jf_agent/pull/248/commits/2bfdbd50c418a9336b588881aebbe9e79c13ce57#diff-b9f32c609e4bd39398d8a39e8f65ae5978f4b9db7a7c023eaa470ee438043373R329

**Pull request merged**